Deselect list item automatically

AmauryMi8 opened the pull request:
when go back to systems collection view, deselect current item.

Half way from success because still can not select same item 2 times...checking why

Commit:
Deselect list item automatically

when go back to systems collection view, deselect current item.

still half way from succes because of can not select same item 2 times...checking why

jeremykairis commented on challengeMk2/ViewModels/StarSystemsViewModel.cs:
```diff
@@ -126,13 +129,15 @@ List<StarSystem> GetOfflineData(string savedFile)
 
             if (File.Exists(savedFile))  // User has already saved datas when he has internet connection
```
Clarify the code to remove comments.

jeremykairis commented on challengeMk2/Views/StarSystemsPage.xaml.cs:
```diff
@@ -31,6 +31,8 @@ protected override void OnAppearing()
 
             var vm = BindingContext as StarSystemsViewModel;
 
+            SystemCollection.SelectedItem = null;
```
What is `//To trigger Auto PullToRefresh collection` ?

jeremykairis commented on challengeMk2/Views/StarSystemsPage.xaml.cs:
```diff
@@ -31,6 +31,8 @@ protected override void OnAppearing()
 
             var vm = BindingContext as StarSystemsViewModel;
 
+            SystemCollection.SelectedItem = null;
```
By the way, selectedItem should be set to null in the setter.

AmauryMi8 replied:
It is set in the setter now and allows to reselect the same item without keep the selection when go back in the list view.

Commit:
fixed collectionView selection

SelectedSystem is changed in the setter
No more selection when go back in the listview and reselect the same item

Commit:
Clean up

Set selectedItem in the setter (really this time...)
and remove it from code behind.
rename DataFile to dbDataFile and remove useless comments.

jeremykairis commented on challengeMk2/ViewModels/StarSystemsViewModel.cs:
```diff
@@ -30,7 +30,11 @@ public StarSystem SelectedSystem
             set
             {
-                if (value != null)
+                if (value == null)
```
Could be improved using the early return pattern.

jeremykairis commented on challengeMk2/Views/StarSystemsPage.xaml.cs:
```diff
@@ -31,7 +31,8 @@ protected override void OnAppearing()
 
-            SystemCollection.SelectedItem = null;
+            //SystemCollection.SelectedItem = null;
+            vm.SelectedSystem = null;
```
It should be removed from here. (Encapsulation)

Commit:
Merge branch 'master' into CollectionSelectionFix
